### PR TITLE
Rework PE wrapper in architecture

### PR DESCRIPTION
**Pull request by cahz**

### CODE REVIEW

File: toolflow/vivado/arch/axi4mm/axi4mm.tcl [hunk 1]
```diff
@@ -375,7 +368,8 @@ namespace eval arch {
   # Connect internal clock lines.
   proc arch_connect_clocks {} {
     connect_bd_net [tapasco::subsystem::get_port "design" "clk"] \
-      [get_bd_pins -of_objects [get_bd_cells] -filter "TYPE == clk && DIR == I"]
+      [get_bd_pins -of_objects [get_bd_cells] -filter "TYPE == clk && DIR == I"] \
```
Comment (wirthjohannes) [inline toolflow/vivado/arch/axi4mm/axi4mm.tcl:371]: What does line 371 do?
The clock ports of the PEs are already connected by the following line

File: toolflow/vivado/arch/axi4mm/axi4mm.tcl [hunk 2]
```diff
@@ -384,8 +378,8 @@ namespace eval arch {
       [get_bd_pins -of_objects [get_bd_cells] -filter "TYPE == rst && NAME =~ *interconnect_aresetn && DIR == I"]
     connect_bd_net [tapasco::subsystem::get_port "design" "rst" "peripheral" "resetn"] \
       [get_bd_pins -of_objects [get_bd_cells -of_objects [current_bd_instance .]] -filter "TYPE == rst && NAME =~ *peripheral_aresetn && DIR == I"] \
-      [get_bd_pins -filter { TYPE == rst && DIR == I && CONFIG.POLARITY != ACTIVE_HIGH } -of_objects [get_bd_cells -filter {NAME =~ "target_ip*"}]]
-    set active_high_resets [get_bd_pins -of_objects [get_bd_cells] -filter "TYPE == rst && DIR == I && CONFIG.POLARITY == ACTIVE_HIGH"]
+      [get_bd_pins -filter { TYPE == rst && DIR == I && CONFIG.POLARITY != ACTIVE_HIGH } -of_objects [get_bd_cells -hier -filter {PATH =~ "*target_ip*"}]]
+    set active_high_resets [get_bd_pins -of_objects [get_bd_cells -hier] -filter "TYPE == rst && DIR == I && CONFIG.POLARITY == ACTIVE_HIGH"]
```
Comment (wirthjohannes) [inline toolflow/vivado/arch/axi4mm/axi4mm.tcl:382]: It seems there are inconsistencies how the relevant bd_cells are selected. When looking at the lines 380, 381 and 382 why are there three different calls to get_bd_cells?

**Review by wirthjohannes — CHANGES_REQUESTED** [review]
There are some inconsistencies with connecting clocks and resets (see comments).

Also there is still issue #256. Other interfaces (e.g. AXI-Stream) should just be forwarded, however I'm not really sure what the best option for this would be.

File: toolflow/vivado/arch/axi4mm/axi4mm.tcl (hunk 1 above)
Comment (cahz) [reply toolflow/vivado/arch/axi4mm/axi4mm.tcl:371]: It is required for connecting the in- and out-interconnects

File: toolflow/vivado/arch/axi4mm/axi4mm.tcl (hunk 2 above)
Comment (cahz) [reply toolflow/vivado/arch/axi4mm/axi4mm.tcl:382]: Same here, first all the top level modules in `/arch`, then all PEs within the wrapper. For ACTIVE_HIGH, we don't have to exclude the interconnect trees.

Comment (wirthjohannes) [reply toolflow/vivado/arch/axi4mm/axi4mm.tcl:382]: Make sense

**Review by wirthjohannes — APPROVED** [review]
LGTM

**Pull request merged**